batch.go

```diff
@@ -0,0 +1,94 @@
+package iavl
+
+import (
+	dbm "github.com/cosmos/cosmos-db"
+)
+
+// BatchWithFlusher is a wraper
+// around batch that flushes batch's data to disk
+// as soon as the configurable limit is reached.
+type BatchWithFlusher struct {
+	db                      dbm.DB    // This is only used to create new batch
+	batch                   dbm.Batch // Batched writing buffer.
+	batchSizeFlushThreshold int       // The maximum size of the batch in bytes before it gets flushed to disk
+}
+
+var _ dbm.Batch = BatchWithFlusher{}
+
+// this value is used to estimate the additional size when adding an entry to a batch
+// additional size = key length + value length + possibleEntryOverHead
+var possibleEntryOverHead = 100
```

Comment:
Is this DB backend specific?

If that's the case, I think we should utilize this depending on the DB backend itself.

Reply:
I'm concerned with exposing DB-backend-specific constraints on all other backends. Ideally, this complexity should be abstracted away under the backend implementation.

Imagine only rocksdb has this `possibleEntryOverhead` constraint and goleveldb doesn't.

Then, for goleveldb we would flush earlier than the threshold is actually reached due to over accounting for the `possibleEntryOverhead` that is rocks-backend specific

Reply:
I see. Discussed offline more - so this only applies to a few backends, however, we overapply this to backends that do not have this overhead only when accounting for the last item in the batch.

As a result of the discussion, I agree to keep the implementation as is. 

Let's please add more detailed documentation explaining all the context, including:
- the backends to which this overhead applies, including links to the relevant parts in the codebase
- the overhead applies to every entry in the batch. However, we over-account for this overhead for backends that do not have this overhead only for the last entry in the batch.